eox_tenant/auth.py

```diff
@@ -0,0 +1,95 @@
+
+"""
+This file implements the authentication backend for the openedx platform.
+"""
+import logging
+import re
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import AllowAllUsersModelBackend as UserModelBackend
+
+from django.utils.translation import ugettext as _
+
+from ratelimitbackend.backends import RateLimitMixin
+
+from eox_tenant.edxapp_wrapper.auth import get_edx_auth_failed
+
+AuthFailedError = get_edx_auth_failed()
+AUDIT_LOG = logging.getLogger("audit")
+UserModel = get_user_model()
+
+
+class TenantAwareAuthBackendBase(UserModelBackend):
+    """
+    Authentication Backend class which will check if the user has a signupsource in the requested site.
+    """
+    def authenticate(self, request, username=None, password=None, **kwargs):
```

Comment:
@morenol I think we can use the `super` method here to avoid replicating the code from django. What we can do is overwriting the user_can_authenticate method and there

Reply:
@jfavellar90 the user_can_authenticate method does not receive the "request" which is being used to get the HTTP_HOST